include/range/v3/view/single.hpp

```diff
@@ -38,11 +39,11 @@ namespace ranges
         {
         private:
             friend struct range_access;
-            Val value_;
+            semiregular_t<Val> value_;
             struct cursor
             {
             private:
-                Val value_;
+                semiregular_t<Val> value_;
```

Comment:
I think you missed this one. This should be `Val` instead of `semiregular_t<Val>`.